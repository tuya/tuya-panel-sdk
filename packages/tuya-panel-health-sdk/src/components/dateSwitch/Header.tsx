import React, { ReactNode } from 'react';
import { View, StyleSheet, TouchableOpacity, TouchableOpacityProps } from 'react-native';

import { IconFont, TYText, Utils } from 'tuya-panel-kit';

const { convertX: cx } = Utils.RatioUtils;

export interface HeaderProps {
  prefixCls?: string;
  title?: ReactNode;
  onPrev?: () => void;
  onNext?: () => void;
  // value?: Dayjs;
  value: string;
  color?: string;
}

interface ButtonProps extends TouchableOpacityProps {
  icon: string;
  color: string;
}

function Button(props: ButtonProps) {
  const { icon = 'arrow', color = '#000', ...other } = props;
  return (
    <TouchableOpacity {...(other as TouchableOpacityProps)}>
      <View style={styles.iconButton}>
        <IconFont name={icon} color={color} />
      </View>
    </TouchableOpacity>
  );
}

function Header(props: HeaderProps) {
  const { title, onPrev, onNext, value, color } = props;
  return (
    <View style={styles.container}>
      <Button icon="backIos" onPress={onPrev} color={color} />
      <View>
        <TYText size={16} color={color}>
          {value}
        </TYText>
      </View>
      <Button icon="arrow" onPress={onNext} color={color} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: cx(10),
    paddingVertical: 8,
  },

  // BUTTON
  iconButton: {
    alignItems: 'center',
    backgroundColor: 'rgba(255,255,255, 0.1)',
    borderRadius: 13,

    height: cx(26),
    justifyContent: 'center',
    width: cx(26),
  },
});

export default Header;
