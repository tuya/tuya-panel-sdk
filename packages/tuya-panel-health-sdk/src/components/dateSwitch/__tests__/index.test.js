import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import { DaySwitch, WeekSwitch, MonthSwitch } from '../index';
import Header from '../Header';

describe('DaySwitch components', () => {
  it('basic DaySwitch render', () => {
    const wrapper = shallow(<DaySwitch value={Date.now()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('basic Header render', () => {
    const wrapper = shallow(<Header value="2022" onPrev={() => {}} onNext={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('basic DaySwitch render controlled', () => {
    const wrapper = shallow(<DaySwitch />);
    expect(wrapper).toMatchSnapshot();
  });

  it('basic WeekSwitch render', () => {
    const wrapper = shallow(
      <WeekSwitch
        value={[Date.now(), Date.now()]}
        onChange={dateString => {
          console.log(dateString);
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('basic MonthSwitch render', () => {
    const wrapper = shallow(
      <MonthSwitch
        value={[Date.now(), Date.now()]}
        onChange={(dateString, date) => {
          console.log(dateString);
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
