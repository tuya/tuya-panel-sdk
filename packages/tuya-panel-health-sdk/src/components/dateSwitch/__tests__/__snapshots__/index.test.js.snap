// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaySwitch components basic DaySwitch render 1`] = `
<View>
  <Header
    onNext={[Function]}
    onPrev={[Function]}
    value="2022-01-18"
  />
</View>
`;

exports[`DaySwitch components basic DaySwitch render controlled 1`] = `
<View>
  <Header
    onNext={[Function]}
    onPrev={[Function]}
    value="2022-01-18"
  />
</View>
`;

exports[`DaySwitch components basic Header render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 20,
      "paddingVertical": 8,
    }
  }
>
  <Button
    icon="backIos"
    onPress={[Function]}
  />
  <View>
    <TYText
      align={null}
      color={null}
      size={16}
      type={null}
      weight={null}
    >
      2022
    </TYText>
  </View>
  <Button
    icon="arrow"
    onPress={[Function]}
  />
</View>
`;

exports[`DaySwitch components basic MonthSwitch render 1`] = `
<View>
  <Header
    onNext={[Function]}
    onPrev={[Function]}
    value="2022-01"
  />
</View>
`;

exports[`DaySwitch components basic WeekSwitch render 1`] = `
<View>
  <Header
    onNext={[Function]}
    onPrev={[Function]}
    value="2022-01-18-2022-01-18"
  />
</View>
`;
